eox_nelp/notifications/admin.py

```diff
@@ -0,0 +1,98 @@
+"""Notifications admin file.
+
+Contains the admin models for notifications.
+
+classes:
+    DueDateListFilter: Admin class for LikeDislikeUnit model.
+    UpcomingCourseDueDateAdmin: Admin class for UpcomingCourseDueDate model.
+
+functions:
+    notify_due_dates: It will call notify method for every queryset record,
+                      this only works with UpcomingCourseDueDate models.
+"""
+import datetime
+
+from django.contrib import admin
+from django.contrib.admin import DateFieldListFilter
+from django.utils import timezone
+from django.utils.translation import gettext_lazy as _
+
+from eox_nelp.notifications.models import UpcomingCourseDueDate
+
+
+class DueDateListFilter(DateFieldListFilter):
+    """Filter class that allows to filter by upcoming dates.
+
+    The filter action will be applied from the current day until the specified parameter,
+    the options are the following:
+
+        1) Any date: it won't apply any filter.
+        2) Next day: Records from now until 1 day in the future.
+        3) Next seven days: Records from now until 7 days in the future.
+        4) Next month: Records from now until 30 days in the future.
+    """
+
+    def __init__(self, field, request, params, model, model_admin, field_path):  # pylint: disable=too-many-arguments
+        """Custom init method to modify inherited options"""
+        super().__init__(field, request, params, model, model_admin, field_path)
+        today = timezone.now()
+
+        self.links = (
+            (_("Any date"), {}),
+            (
+                _("Next day"),
+                {
+                    self.lookup_kwarg_since: today,
+                    self.lookup_kwarg_until: today + datetime.timedelta(days=1),
+                },
+            ),
+            (
+                _("Next seven days"),
+                {
+                    self.lookup_kwarg_since: today,
+                    self.lookup_kwarg_until: today + datetime.timedelta(days=7),
+                },
+            ),
+            (
+                _("Next month"),
+                {
+                    self.lookup_kwarg_since: today,
+                    self.lookup_kwarg_until: today + datetime.timedelta(days=30),
+                },
```

Comment:
What do you think of adding more than 30 days filter or its innecessary?

Reply:
Could you give me an example, where that could be necessary ?

Reply:
No looking more the filters are not Mutual exclusivity so exists intersection between the next day and the next week for example.
So for that case, I was thinking of filtering by more than one month but not including between today and the month.  But seems to be not the purpose of the filter, so forget it.